lib/nxdk/configsector.c

```diff
@@ -147,3 +148,66 @@ bool nxLoadNetworkConfig(nxdk_network_config_sector_t *config)
 
     return true;
 }
+
+bool nxSaveNetworkConfigSector(HANDLE hPartition, UINT sectorIndex, const BYTE *buffer, UINT size)
+{
+    NXDK_CONFIG_SECTOR configSector;
+
+    _Static_assert(sizeof(configSector) == 512, "NXDK_CONFIG_SECTOR size mismatch");
+
+    assert(hPartition != INVALID_HANDLE_VALUE);
+    if (hPartition == INVALID_HANDLE_VALUE) {
+        return false;
+    }
+
+    assert(sectorIndex < NXDK_CONFIG_SECTOR_COUNT);
```

Comment:
assert here, then a check just below too? Should we remove the assert?

Reply:
That's on purpose - if this fails, you have a bug in your code, so when it happens in a non-debug build, it should gracefully fail, but in a debug build, I prefer stuff problems like these to be put right in my face so I can't ignore them.